src/test/scala/io/iohk/ethereum/blockchain/sync/RegularSyncSpec.scala

```diff
@@ -373,32 +372,31 @@ class RegularSyncSpec extends TestKit(ActorSystem("RegularSync_system")) with Wo
         sendBlockHeadersFromBlocks(Seq(newBlock))
         sendBlockBodiesFromBlocks(Seq(newBlock))
 
-        Thread.sleep(1000)
-
-        regularSync.underlyingActor.missingStateNodeRetry shouldEqual
-          Some(MissingStateNodeRetry(missingNodeHash, peer1, Seq(newBlock)))
+        time.advance(1000)
 
         sendNodeData(Seq(missingNodeValue))
-        Thread.sleep(1000)
-
-        regularSync.underlyingActor.missingStateNodeRetry shouldEqual None
+        eventually {
+          regularSync.underlyingActor.isBlacklisted(peer1.id) shouldBe false
+        }
       }
     }
   }
 
   trait TestSetup extends DefaultSyncConfig with EphemBlockchainTestSetup with SecureRandomBuilder {
-    override implicit lazy val system: ActorSystem = testkitActorSystem
+    override implicit lazy val system: ActorSystem = testKitActorSystem
+
+    val time = new VirtualTime
```

Comment:
`time.advance` won't be working, because you don't pass the `time` to the actor. You are passing `system.scheduler` instead of `time.scheduler`